connect-inject/endpoints_controller.go

```diff
@@ -210,6 +228,111 @@ func (r *EndpointsController) SetupWithManager(mgr ctrl.Manager) error {
 		).Complete(r)
 }
 
+// getServiceCheck will return the health check for this pod and service if it exists.
+func getServiceCheck(client *api.Client, healthCheckID string) (*api.AgentCheck, error) {
+	filter := fmt.Sprintf("CheckID == `%s`", healthCheckID)
+	checks, err := client.Agent().ChecksWithFilter(filter)
+	if err != nil {
+		return nil, err
+	}
+	// This will be nil (does not exist) or an actual check.
+	return checks[healthCheckID], nil
+}
+
+// registerConsulHealthCheck registers a TTL health check for the service on this Agent.
+// The Agent is local to the Pod which has a kubernetes health check.
```

Comment:
nit: the pod may not have a k8s health check, but it will have a Ready status which is what we're actually sync'ing.

Reply:
Updated this to use the phrase "readiness status", WDYT?

Reply:
Love it!